spring-restdocs-core/src/main/java/org/springframework/restdocs/ManualRestDocumentation.java

```diff
@@ -31,17 +33,31 @@
  */
 public final class ManualRestDocumentation implements RestDocumentationContextProvider {
 
+	private static final String GENERATED_SNIPPETS_PATH = "generated-snippets";
+	private static final String MAVEN_TARGET_PATH = "target" + File.separator + GENERATED_SNIPPETS_PATH;
+	private static final String GRADLE_BUILD_PATH = "build" + File.separator + GENERATED_SNIPPETS_PATH;
+	private static final String MAVEN_POM = "pom.xml";
+	private static final String OUTPUT_DIR_PROPERTY_KEY = "snippetOutputDirectory";
+
 	private final File outputDirectory;
 
 	private RestDocumentationContext context;
 
 	/**
 	 * Creates a new {@code ManualRestDocumentation} instance that will generate snippets
+	 * to &lt;gradle/maven build path&gt;/generated-snippet if no runtime property ({@code snippetOutputDirectory}) is set.
+	 */
+	public ManualRestDocumentation() {
+		this(System.getProperty(OUTPUT_DIR_PROPERTY_KEY, getDefaultOutputDirectory()));
```

Comment:
I don't like the use of a system property.

REST Docs used one in the past but it didn't work well when running individual tests in an IDE. Typically, the system property is only set in `build.gradle` or `pom.xml` so you ended up with different configuration.

For now at least, I'm happy for people who don't want to use the conventional locations to have to override the location using the constructor argument to `JUnitRestDocumentation` or `ManualRestDocumentation` and continuing to use `include::`.